### PR TITLE
build: update ascend dockerfile

### PR DESCRIPTION
**Pull request by CyCle1024**
## Motivation

Using dlinfer instead of deeplink.framework in Dockerfile build for device=`ascend`

## Modification

Update Dockerfile_aarch64_ascend, using `pip3 install dlinfer-ascend==0.1.0` to support `ascend`,
`torch_npu` is used internally.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docker/Dockerfile_aarch64_ascend [hunk 1]
```diff
@@ -115,57 +103,25 @@ SHELL ["/bin/bash", "-c"]
...
+    pip3 install dlinfer-ascend==0.1.0
 
 # lmdeploy
-WORKDIR /workspace
-RUN git clone https://github.com/InternLM/lmdeploy.git && \
-    cd lmdeploy && \
-    git checkout ${LMDEPLOY_TAG_OR_COMMIT} && \
+WORKDIR /opt/lmdeploy
+
+COPY autotest /opt/lmdeploy/autotest
```
Comment (lvhan028) [inline docker/Dockerfile_aarch64_ascend:116]: I understand that `src` has nothing to do with ascend inference. 
But user might modify code and would like to commit to github. 
So please keep it as a whole package
```
COPY . /opt/lmdeploy

WORKDIR /opt/lmdeploy

```

Comment (lvhan028) [inline docker/Dockerfile_aarch64_ascend:116]: where autotest comes from?

Comment (CyCle1024) [reply docker/Dockerfile_aarch64_ascend:116]: image
Red rectangle selected region in picture, just a dir under repo root dir.

Comment (CyCle1024) [reply docker/Dockerfile_aarch64_ascend:116]: `COPY . /opt/lmdeploy` could bring all ascend `*.run` files into docker images (about 3.8GB extra size for CANN 8.0.RC3.alpha001), I could not find a good way to exclude the ascend  files.
Actually, there is an experimental syntax for Dockerfile that excludes certain patterns [link](https://docs.docker.com/reference/dockerfile/#copy---exclude)
```Dockerfile
COPY --excludes *.run . /opt/lmdeploy
```
It's experimental and I have not tested yet.

Comment (CyCle1024) [reply docker/Dockerfile_aarch64_ascend:116]: I think i

> I understand that `src` has nothing to do with ascend inference. But user might modify code and would like to commit to github. So please keep it as a whole package
> 
> ```
> COPY . /opt/lmdeploy
> 
> WORKDIR /opt/lmdeploy
> ```

I think it's ok if extra 3.8GB size for ascend image is acceptable.

Comment (CyCle1024) [reply docker/Dockerfile_aarch64_ascend]: @lvhan028 using a new multi-stage build avoid copy run file into final image, please review

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**